Add control on dapps endpoint to handle null data

### Expected behaviour

Tests always pass.

### Actual behaviour
After introducing transactions on the database through some of the functional POST test, we obtain next error: 
```
 7 failing

  1) GET /dapps
       /
         should return all results:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.9.icon | Expected type string but found type null
INVALID_TYPE: data.8.tags | Expected type string but found type null
INVALID_TYPE: data.1.tags | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at process._tickCallback (internal/process/next_tick.js:109:7)

  2) GET /dapps
       ?
         limit=
           using 100 should be ok:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.11.description | Expected type string but found type null
INVALID_TYPE: data.10.description | Expected type string but found type null
INVALID_TYPE: data.9.icon | Expected type string but found type null
INVALID_TYPE: data.8.tags | Expected type string but found type null
INVALID_TYPE: data.1.tags | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)

  3) GET /dapps
       ?
         limit=1&
           using offset 1 should be ok:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.0.tags | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)

  4) GET /dapps
       ?
         offset=
           using offset 0 should return different result than offset 1:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.9.description | Expected type string but found type null
INVALID_TYPE: data.8.icon | Expected type string but found type null
INVALID_TYPE: data.7.tags | Expected type string but found type null
INVALID_TYPE: data.0.tags | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)

  5) GET /dapps
       ?
         sort=
           using "name:desc" should return result in descending order:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.5.description | Expected type string but found type null
INVALID_TYPE: data.1.icon | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)

  6) GET /dapps
       ?
         unknown=
           using empty string should return all results:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.9.icon | Expected type string but found type null
INVALID_TYPE: data.8.tags | Expected type string but found type null
INVALID_TYPE: data.1.tags | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)

  7) GET /dapps
       ?
         unknown=
           using "unknown" should return all results:
     AssertionError: InvalidResponseBody: INVALID_TYPE: data.9.icon | Expected type string but found type null
INVALID_TYPE: data.8.tags | Expected type string but found type null
INVALID_TYPE: data.1.tags | Expected type string but found type null
      at test/common/swaggerSpec.js:184:22
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)
```

### Steps to reproduce

Run functional post dapps related tests and afterwards run functional test get dapps over the same database.

--- a/api/controllers/dapps.js
+++ b/api/controllers/dapps.js
@@ -36,6 +36,21 @@ DappsController.getDapps = function (context, next) {
 		try {
 			if (err) { return next(err); }
 
+			data = _.cloneDeep(data);
+
+			data = _.map(data, function (dapp) {
+				if (_.isNull(dapp.description)) {
+					dapp.description = '';
+				}
+				if (_.isNull(dapp.tags)) {
+					dapp.tags = '';
+				}
+				if (_.isNull(dapp.icon)) {
+					dapp.icon = '';
+				}
+				return dapp;
+			});
+
 			next(null, {
 				data: data,
 				meta: {

